### PR TITLE
Document total ad blocking time audit

### PR DESCRIPTION
**Pull request by warrengm**

### CODE REVIEW

File: docs/audits/total-ad-blocking-time.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+attributable to ads scripts. This captures time where the JS thread is blocked
+by an ad script for long enough to be noticeable by users when it impacts
+input delay or content loading.
+
+## Recommendations
+
+The most important factor in ad JS blocking time is how many ads are being
+loaded. Our recommendations are:
+
+- Load fewer ads or reduce ad density, if feasible
```
Comment (jburger424) [inline docs/audits/total-ad-blocking-time.md:15]: Link to ad density docs

File: docs/audits/total-ad-blocking-time.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+- Lazy load ads below the fold
```
Comment (jburger424) [inline docs/audits/total-ad-blocking-time.md:16]: Link to lazy load API docs

File: docs/audits/total-ad-blocking-time.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+- Reduce batch sizes if using Single Request Architecture (SRA) in GPT.
```
Comment (jburger424) [inline docs/audits/total-ad-blocking-time.md:17]: Link to docs explaining how to do this

**Review by jburger424 — APPROVED** [review]

File: docs/audits/total-ad-blocking-time.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Reduce Ad JS Blocking Time
+
+## Overview
+
+This audit measures how much [blocking time](https://web.dev/tbt/) is
+attributable to ads scripts. This captures time where the JS thread is blocked
```
[the remark this answers is not in the document]
Comment (jimper) [reply docs/audits/total-ad-blocking-time.md:6]: +1 ad-related scripts

File: docs/audits/total-ad-blocking-time.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+by an ad script for long enough to be noticeable by users when it impacts
```
Comment (jimper) [inline docs/audits/total-ad-blocking-time.md:7]: There should be a "More Information" section at the end that includes:

1. What constitutes "long enough" here
2. What ad scripts are supported

Should also probably call out when this measurement starts/stops.

File: docs/audits/total-ad-blocking-time.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Reduce Ad JS Blocking Time
+
+## Overview
+
+This audit measures how much [blocking time](https://web.dev/tbt/) is
```
Comment (jimper) [inline docs/audits/total-ad-blocking-time.md:5]: Suggest rewording to: "The amount of blocking time attributable to ad-related scripts."

File: docs/audits/total-ad-blocking-time.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+attributable to ads scripts. This captures time where the JS thread is blocked
+by an ad script for long enough to be noticeable by users when it impacts
+input delay or content loading.
+
+## Recommendations
+
+The most important factor in ad JS blocking time is how many ads are being
+loaded. Our recommendations are:
```
Comment (jimper) [inline docs/audits/total-ad-blocking-time.md:13]: Isn't it more *when* ads are being loaded, rather than the overall number? If so, maybe reword to "...how many ads are being loaded at once" or something.

File: docs/audits/total-ad-blocking-time.md (hunk 1 above)
Comment (jimper) [reply docs/audits/total-ad-blocking-time.md:15]: https://developers.google.com/publisher-ads-audits/reference/audits/viewport-ad-density

File: docs/audits/total-ad-blocking-time.md (hunk 2 above)
Comment (jimper) [reply docs/audits/total-ad-blocking-time.md:16]: https://developers.google.com/publisher-tag/samples/lazy-loading might be more useful

File: docs/audits/total-ad-blocking-time.md (hunk 3 above)
Comment (jimper) [reply docs/audits/total-ad-blocking-time.md:17]: https://developers.google.com/publisher-tag/guides/publisher-console-messages#TOO_MANY_SLOTS_IN_SRA_REQUEST

**Review by jimper — CHANGES_REQUESTED** [review]

File: docs/audits/total-ad-blocking-time.md (hunk 3 above)
Comment (warrengm) [reply docs/audits/total-ad-blocking-time.md:17]: Done -- but I wonder if it's easier to use the refresh API for custom batching

File: docs/audits/total-ad-blocking-time.md (hunk 7 above)
Comment (warrengm) [reply docs/audits/total-ad-blocking-time.md:13]: Done. Although I imagine there is some correlation with the total number of ads even if they're split into small batches

File: docs/audits/total-ad-blocking-time.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+- [Reduce batch sizes if using Single Request Architecture (SRA) in GPT](https://developers.google.com/publisher-tag/guides/publisher-console-messages#TOO_MANY_SLOTS_IN_SRA_REQUEST).
+
+## More information
+
+The metric sums the _blocking time_ of each JS task coming from an ad-related
+script. See [web.dev](https://web.dev/lighthouse-total-blocking-time/#what-tbt-measures)
+for more information about this is defined.
+
+A script is considered to be ad-related if it is either an ad tag or labeled as
+an "ad" script by https://github.com/patrickhulce/third-party-web.
```
Comment (jimper) [inline docs/audits/total-ad-blocking-time.md:26]: `[Third Party Web](https://github.com/patrickhulce/third-party-web)`

**Review by jburger424 — APPROVED** [review]

File: docs/audits/total-ad-blocking-time.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+loaded. Our recommendations are:
+
+- [Load fewer ads at once or reduce ad density](https://developers.google.com/publisher-ads-audits/reference/audits/viewport-ad-density), if feasible
+- [Lazy load ads below the fold](https://developers.google.com/publisher-ads-audits/reference/audits/ads-in-viewport)
+- [Reduce batch sizes if using Single Request Architecture (SRA) in GPT](https://developers.google.com/publisher-tag/guides/publisher-console-messages#TOO_MANY_SLOTS_IN_SRA_REQUEST).
+
+## More information
+
+The metric sums the _blocking time_ of each JS task coming from an ad-related
+script. See [web.dev](https://web.dev/lighthouse-total-blocking-time/#what-tbt-measures)
```
Comment (jimper) [inline docs/audits/total-ad-blocking-time.md:22]: `To learn how blocking time is defined, see this article on [Total Blocking Time](https://web.dev/lighthouse-total-blocking-time/#what-tbt-measures).`

File: docs/audits/total-ad-blocking-time.md (hunk 7 above)
Comment (jimper) [reply docs/audits/total-ad-blocking-time.md:13]: I don't see any change here, but I'm fine leaving as-is if you think the number of ads is actually the most important factor.

File: docs/audits/total-ad-blocking-time.md (hunk 8 above)
Comment (jimper) [reply docs/audits/total-ad-blocking-time.md:26]: Also, is this different from say https://developers.google.com/publisher-ads-audits/reference/audits/ad-render-blocking-resources#more_information?

**Review by jimper — CHANGES_REQUESTED** [review]

File: docs/audits/total-ad-blocking-time.md (hunk 8 above)
Comment (warrengm) [reply docs/audits/total-ad-blocking-time.md:26]: Yes, it's different

**Review by jimper — APPROVED** [review]
